### PR TITLE
refactor(NODE-3845): update load balancer config

### PR DESCRIPTION
**Pull request by durran**
### Description

Updates the driver to properly use load balancers against latest server.

#### What is changing?

Removes the use of `FAKE_MONGODB_SERVICE_ID` in the code and now tests against the latest server with load balancers. Requires https://github.com/mongodb-labs/drivers-evergreen-tools/pull/183 to be merged first.

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

NODE-3845 / DRIVERS-1983

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/config.yml.in
Comment (durran) [inline .evergreen/config.yml.in:90]: This change must be reverted before merging - only here to make the tests pass since that branch getting merge is a requirement for this PR.

File: test/integration/connection-monitoring-and-pooling/connection.test.js
Comment (nbbeeken) [inline test/integration/connection-monitoring-and-pooling/connection.test.js:19]: https://github.com/mongodb/node-mongodb-native/blob/main/test/tools/runner/filters/mongodb_topology_filter.js

Unfortunately this "not" logic isn't implemented in the topology filter, so this is just filtering on all topologies since it does exact string matches. Maybe this is a good time to add it. The os_filter does support this so we could use the same logic from there.

File: .evergreen/config.yml.in
Comment (nbbeeken) [reply .evergreen/config.yml.in:90]: Maybe we could merge this if we get it all ready to go in, and then when the tools PR is merged we can just do a oneliner to make this go back to pulling from master. As long as we expect it to go in soon. My thinking is we're waiting to hear on when all the teams want to contend with the failures that they'll all start seeing when this goes in. 

We might as well proceed with our fix if that makes sense? (@dariakp @baileympearson)

Comment (dariakp) [reply .evergreen/config.yml.in:90]: That's not a terrible idea, but then we should file a follow up and schedule it immediately to make sure it doesn't get lost

File: test/integration/connection-monitoring-and-pooling/connection.test.js
Comment (durran) [reply test/integration/connection-monitoring-and-pooling/connection.test.js:19]: I've added to the topology filter now.

File: .evergreen/config.yml.in
Comment (durran) [reply .evergreen/config.yml.in]: Drivers tools has been merged so this change has now been reverted.

**Review by nbbeeken — DISMISSED** [review]

**Review by dariakp — CHANGES_REQUESTED** [review]
2 comments, one regarding the filter below, the other is - since this is internal, should we label is at `feat:` (i.e. mandatory semver bump), or could use `refactor:`?

File: test/tools/runner/filters/mongodb_topology_filter.js
Comment (nbbeeken) [reply test/tools/runner/filters/mongodb_topology_filter.js:43]:
```suggestion
    const isExclusion = topologies[0][0] === '!';
    if (isExclusion) {
      if (!topologies.every(topology => topology.startsWith('!'))) {
        // Not every topology starts with !
        throw new Error('Cannot combine exclusion with inclusion of topologies');
      }

      // Every excluded topology does not equal the current (prefix !)
      return topologies.every(topology => topology !== `!${this.runtimeTopology}`);
    } else {
      // inclusion list
      if (topologies.some(topology => topology.startsWith('!'))) {
        // Some topologies start with !
        throw new Error('Cannot combine exclusion with inclusion of topologies');
      }

      // At least some (one) of the included topologies equals the current
      return topologies.some(topology => topology === this.runtimeTopology);
    }
```

This boolean logic threw me for a loop, I broke it up into a commented example for my own sanity

**Review by dariakp — APPROVED** [review]

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**
